spring-pulsar/src/test/java/org/springframework/pulsar/core/PulsarTemplateTests.java

```diff
@@ -60,18 +61,67 @@
  */
 class PulsarTemplateTests extends AbstractContainerBaseTests {
 
-	private static final String SAMPLE_MESSAGE_KEY = "sample-key";
+	@ParameterizedTest(name = "{0}")
+	@MethodSource("interceptorInvocationTestProvider")
+	void interceptorInvocationTest(String topic, List<ProducerInterceptor> interceptors) throws Exception {
+		try (PulsarClient client = PulsarClient.builder().serviceUrl(getPulsarBrokerUrl()).build()) {
+			PulsarProducerFactory<String> producerFactory = new DefaultPulsarProducerFactory<>(client,
+					Collections.singletonMap("topicName", topic));
+			PulsarTemplate<String> pulsarTemplate = new PulsarTemplate<>(producerFactory, interceptors);
+			pulsarTemplate.send("test-interceptor");
+			for (ProducerInterceptor interceptor : interceptors) {
+				verify(interceptor, atLeastOnce()).eligible(any(Message.class));
+			}
+		}
+	}
 
-	private static final TypedMessageBuilderCustomizer<String> sampleMessageKeyCustomizer = messageBuilder -> messageBuilder
-			.key(SAMPLE_MESSAGE_KEY);
+	private static Stream<Arguments> interceptorInvocationTestProvider() {
+		return Stream.of(
+				arguments(Named.of("testSingleInterceptor", "iit-topic-1"),
+						Collections.singletonList(mock(ProducerInterceptor.class))),
+				arguments(Named.of("testMultipleInterceptors", "iit-topic-2"),
+						List.of(mock(ProducerInterceptor.class), mock(ProducerInterceptor.class))));
+	}
+
+	@Test
+	void sendMessageWithSpecificSchemaTest() throws Exception {
+		String topic = "smt-specific-schema-topic";
+		try (PulsarClient client = PulsarClient.builder().serviceUrl(getPulsarBrokerUrl()).build()) {
+			try (Consumer<Foo> consumer = client.newConsumer(Schema.JSON(Foo.class)).topic(topic)
+					.subscriptionName("test-specific-schema-subscription").subscribe()) {
+				PulsarProducerFactory<Foo> producerFactory = new DefaultPulsarProducerFactory<>(client,
+						Collections.singletonMap("topicName", topic));
+				PulsarTemplate<Foo> pulsarTemplate = new PulsarTemplate<>(producerFactory);
+				pulsarTemplate.setSchema(Schema.JSON(Foo.class));
+				Foo foo = new Foo("Foo-" + UUID.randomUUID(), "Bar-" + UUID.randomUUID());
+				pulsarTemplate.send(foo);
+				assertThat(consumer.receiveAsync()).succeedsWithin(Duration.ofSeconds(3)).extracting(Message::getValue)
+						.isEqualTo(foo);
+			}
+		}
+	}
 
 	@ParameterizedTest(name = "{0}")
 	@MethodSource("sendMessageTestProvider")
-	void sendMessageTest(String topic, Map<String, Object> producerConfig, SendHandler<Object> handler,
```

Comment:
Simplified to pass in a `SendTestArgs` builder object instead of some params, then the "SendHandler", it was a bit messy before. 

The reason for using a params builder is to have easy to read test params that look like:
```java
	arguments("sendMessageToDefaultTopicWithAllOptions",					 
        SendTestArgs
            .useSpecificTopic(false)
            .useCustomRouter(true)
            .useMessageCustomizer(true)
            .useProducerCustomizer(true)),
```
rather than a bunch of random booleans that are really hard to easily see which is what and easy to make mistakes with such as: 
```java
	arguments("sendMessageToDefaultTopicWithAllOptions",					 
            false,
            true,
            true, 
            true),
```